registry/handlers/httperror.go

```diff
@@ -0,0 +1,34 @@
+package handlers
+
+import (
+	"github.com/docker/distribution/registry/api/v2"
+	"net/http"
+)
+
+// HTTPError is a wrapper for v2.Errors that adds a Status int to hold
+// an HTTP Status code which will be used to set the status code on
+// the response.
+type HTTPError struct {
+	v2.Errors
+	Status int
+}
+
+// NewHTTPError create a new HTTPError using the given ErrorCode, detail and http status.
+// detail must be json serializable.
+func NewHTTPError(errCode v2.ErrorCode, detail interface{}, status int) *HTTPError {
```

Comment:
Passing by pointer allows me to return nil when there's no error giving slightly cleaner code. Also, if the list of Errors ever becomes large, pass by value will be inefficient.